tabs/src/components/sample/SampleQueries.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect } from "react";
+import { Table, TableCell, OpenOutsideIcon, Button} from '@fluentui/react-northstar';
+import { GRAPH_URL, SAMPLE_QUERIES_URL} from "./TabConstants";
+import * as microsoftTeams from "@microsoft/teams-js";
+import PropTypes from 'prop-types';
+
+FetchSamples.propTypes = {
+    query: PropTypes.string,
+    setQuery: PropTypes.func
+};
+
+export function FetchSamples(props){
+    const [samples, setSamples] = useState([]);
+    useEffect(() => {
+        async function getSamplesList() {
+            microsoftTeams.getContext(async function (context) {
+                const devxApi = SAMPLE_QUERIES_URL;
+
+                const headers = {
+                    'Content-Type': 'application/json'
+                };
+    
+                try {
+                        const response = await fetch(devxApi, headers);
+                        if (!response.ok) {
+                            return;
+                    }
+                    const res = await response.json();
+                    var arr = [];
+                    for(let i = 0; i < res.teamsAppSampleQueries.length; i++){
+
+                        //Filter through all the chats and teams such that it matches the context that this app is attached to
+                        if( (context?.groupId && res.teamsAppSampleQueries[i]["requestUrl"].includes("chat"))||(context?.chatId &&  res.teamsAppSampleQueries[i]["requestUrl"].includes("teams")) ){
+                            continue;
+                        }
+
+                        //Each sample query requires a row
+                        var query = createTableRow(props.setQuery, res.teamsAppSampleQueries[i]["requestUrl"].slice(1,));
+                        query.key = res.teamsAppSampleQueries[i]["id"];
+                        var label = <TableCell className = {"table-cell"} content = {res.teamsAppSampleQueries[i]["method"]}/>;
+                        var button = <TableCell className = {"table-cell"} content = {createFillIn(res.teamsAppSampleQueries[i]["docLink"])}/>;
+                        query.items = [label, res.teamsAppSampleQueries[i]["humanName"], button];
+                        arr.push(query);
+                    }
+                } catch (error) {
+                    return error;
+                }
+                setSamples(arr);
+            });
+        }
+        getSamplesList(samples);
+        
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
dependency array is optional (https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects) if possible, can this be removed? or follow what the linting error is suggesting: React Hook useEffect has missing dependencies: 'props' and 'samples'. Either include them or remove the dependency array. However, 'props' will change when *any* prop changes, so the preferred fix is to destructure the 'props' object outside of the useEffect call and refer to those specific props inside useEffect  react-hooks/exhaustive-deps

Reply:
I remember trying both of these ways before too, but they introduce infinite rendering so I don't think it is a good idea. The reason that this is an empty list is because I only wanted to run when it [mounts and unmounts](https://stackoverflow.com/questions/53070970/infinite-loop-in-useeffect)

Reply:
okay then we can keep the comment to tell eslint to ignore, if there's not way to add the missing dependencies

Reply:
https://gosink.in/react-js-how-to-render-useeffect-only-once/

yeah just keep the []